extends ColorRect

var textBox
var playerRef

const stoneTablet1_content = """
â„¸ Ì£ â‘á’· â„¸ Ì£ âˆ·á”‘â•â„¸ Ì£ ğ™¹âˆ·á“­ ê–á’·â“â„¸ Ì£  âšá“­ â†¸á’·á”‘â†¸
En lâ€™an 131 avant AC, une guerre terrible eu lieu entre les Atlantes et une race extraterrestre venue dâ€™ailleurs. Bien que les Atlantes ont gagnÃ©, plusieurs colonies furent perdu. Une de ces colonies rÃ©ussis quand mÃªme Ã  survivre et Ã  se rÃ©fugier sur une Ã®le. Ils envoyÃ¨rent des messages de dÃ©tresse, mais personne ne vint jamais Ã  leur secours.
"""

const stoneTablet2_content = """
Ê–âšâ„¸ Ì£  âˆ´á’· â“ğ™¹âšãƒªâ†¸ â“á”‘â•â„¸ Ì£ â‘ â•ãƒª aá’·á“­â•âˆ·
Pris de rage de par la lÃ¢chetÃ© de leur peuple, ils dÃ©cidÃ¨rent de bÃ¢tir leur propre citÃ© ainsi que leur propre religion. Ils voulurent par la suite se venger des Atlantes. Pour cela, grÃ¢ce Ã  un sort trÃ¨s puissant, ils crÃ©Ã¨rent un donjon, semblant infini. Quiconque arrivera au sommet sera dotÃ© dâ€™un pouvoir incommensurable.
"""

const stoneTablet3_content = """
á”‘ãƒªâ†¸ aá’·á“­â•âˆ· âˆ´â•ê–ê– á”‘âŠá’·ãƒªâŠ£á’· âšá“­ â•ãƒª â„¸ Ì£ â‘á’· ãƒªá”‘á’²á’· ğ™¹â“ á”‘ê–ê– â‘â•á“­ â“á”‘â•â„¸ Ì£ â‘â“âšê–
Les trois tablettes contiennent chacune une partie de lâ€™invocation dâ€™Aesir, quiconque possÃ¨de les trois peut lâ€™invoquer et ainsi venger cette colonie des Atlantes. Mais Ã  quel prix ?
"""

const diary_content = """
Je suis un Atlante qui a cherchÃ© Ã  se venger toute sa vie. Jâ€™ai cherchÃ© par tous les moyens Ã  rÃ©cupÃ©rer les trois tablettes dâ€™invocation, mais en vain. Je nâ€™Ã©tais pas Ã  la hauteur. Si vous lisez ceci, câ€™est que vous cherchez la mÃªme chose que moi et que vous avez probablement rÃ©ussi. Câ€™est pourquoi je vous fais don de mon savoir Ã  travers ces pages. Maintenant que vous avez dÃ©truit les Atlantes, vous devez penser avoir accompli votre devoir. Mais il nâ€™en est rien. Il nâ€™y a jamais eu de trahison. La guerre a bien eu lieu elle, mais il nâ€™y a pas eu dâ€™abandon. Nous avons juste fui, et avons cachÃ©s notre culpabilitÃ© avec une fausse histoire de trahison. Vous qui avez invoquÃ© Aesir, vous nâ€™avez causÃ© que la perte de votre peuple. Et maintenant la terre entiÃ¨re va en pÃ©rir. Câ€™est bien dommage de finir comme Ã§a. Mais que voulez vous, Ã  part remonter dans le temps, faire machine arriÃ¨re. Il est impossible de changer cela. Parfois il est bien que les secrets restent secrets.
"""

"""
# Called when the node enters the scene tree for the first time.
func _ready():
	#textBox = find_child("TextContent")
"""

func getRefOfDialogue():
	if textBox == null:
		textBox = playerRef.get_node("DialogueBox/TextContent")

func displayBox():
	playerRef.get_node("DialogueBox").show()

func closeBox():
	playerRef.get_node("DialogueBox").hide()

func display_stone1():
	getRefOfDialogue()
	textBox.text = (stoneTablet1_content)  
	displayBox()

func display_stone2():
	getRefOfDialogue()
	textBox.text = (stoneTablet2_content)  
	displayBox()

func display_stone3():
	getRefOfDialogue()
	textBox.text = (stoneTablet3_content)   
	displayBox()

func display_diary():
	getRefOfDialogue()
	textBox.text = (diary_content)  
	displayBox()

func setPlayerRef(node: Node):
	playerRef = node

# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	pass
